### PR TITLE
Combining stencils in tracer_2d_1l.py

### PR DESCRIPTION
**Pull request by gmao-ckung**
## Purpose

The purpose is to combine stencils in `tracer_2d_1l.py` (and `updatedzd.py`).  The following stencils are combined.

- `q_adjust` and `q_other_adjust` are combined into `q_adjustments` in `tracer_2d_1l.py`
- `ra_x_stencil` and `ra_y_stencil` are combined into `ra_stencil_update` in `updatedzd.py`
- `flux_x` and `flux_y` are combined into `flux_compute` in `tracer_2d_1l.py`

## Code changes:

- `q_adjustment` stencil created in `tracer_2d_1l.py`
- `q_other_adjust` stencil deleted from `tracer_2d_1l.py`
- `ra_stencil_update` stencil created in `updatedzd.py`
- `ra_x_stencil` and `ra_y_stencil` stencil definitions are deleted from `updatedzd.py`.
- Calls to `ra_x_stencil` and `ra_y_stencil` in `tracer_2d_1l.py` and `updatedzd.py` are replaced by `ra_stencil_update` 
- `flux_x` and `flux_y` stencils are converted into `gtscript.function` calls in `tracer_2d_1l.py`
- `flux_compute` stencil created in `tracer_2d_1l.py` that call `flux_x` and `flux_y` functions
- `utils.sd` type replaced by `FloatField`

### CODE REVIEW

File: fv3core/stencils/tracer_2d_1l.py [hunk 1]
```diff
@@ -105,25 +163,22 @@ def compute(comm, tracers, dp1, mfxd, mfyd, cxd, cyd, mdt, nq):
...
+        grid.dy,
+        grid.sin_sg1,
         grid.sin_sg2,
+        grid.sin_sg3,
+        grid.sin_sg4,
+        xfx,
         yfx,
-        origin=grid.compute_origin(add=(-grid.halo, 0, 0)),
-        domain=grid.domain_shape_compute(add=(2 * grid.halo, 1, 0)),
+        origin=grid.compute_origin(add=(-grid.halo, -grid.halo, 0)),
```
Comment (rheacangeo) [inline fv3core/stencils/tracer_2d_1l.py:180]: This should be grid.full_origin()

File: fv3core/stencils/tracer_2d_1l.py [hunk 2]
```diff
@@ -105,25 +163,22 @@ def compute(comm, tracers, dp1, mfxd, mfyd, cxd, cyd, mdt, nq):
...
+        domain=grid.domain_shape_compute(add=(2 * grid.halo, 2 * grid.halo, 0)),
```
Comment (rheacangeo) [inline fv3core/stencils/tracer_2d_1l.py:181]: grid.domain_shape_full()

File: fv3core/stencils/tracer_2d_1l.py [hunk 3]
```diff
@@ -176,19 +231,14 @@ def compute(comm, tracers, dp1, mfxd, mfyd, cxd, cyd, mdt, nq):
...
-        domain=grid.domain_shape_compute(add=(0, 2 * grid.halo, 0)),
-    )
-    ra_y_stencil(
-        grid.area,
         yfx,
         ra_y,
-        origin=grid.compute_origin(add=(-grid.halo, 0, 0)),
-        domain=grid.domain_shape_compute(add=(2 * grid.halo, 0, 0)),
+        origin=grid.compute_origin(add=(-grid.halo, -grid.halo, 0)),
+        domain=grid.domain_shape_compute(add=(2 * grid.halo, 2 * grid.halo, 0)),
```
Comment (rheacangeo) [inline fv3core/stencils/tracer_2d_1l.py:241]: same comment about using the 'full' convention

**Review by rheacangeo — COMMENTED** [review]
Looks great! Just a couple of minor comments about origin and domain. I am wondering if qn2 can be preallocated outside of the loop, but not necessary as part of this task. Great job!

File: fv3core/stencils/tracer_2d_1l.py (hunk 2 above)
Comment (gmao-ckung) [reply fv3core/stencils/tracer_2d_1l.py:181]: Fixed!

File: fv3core/stencils/tracer_2d_1l.py (hunk 1 above)
Comment (gmao-ckung) [reply fv3core/stencils/tracer_2d_1l.py:180]: Fixed!

**Comment by gmao-ckung** [comment]
Thanks again for the review!  I made the adjustments about using the "full" origin and domain and updated the branch.  I'll also check into whether qn2 can be moved outside of the loop.  At first glance, it looks like it should be possible.

**Review by rheacangeo — APPROVED** [review]

**Pull request merged**